README.md

```diff
@@ -31,72 +14,74 @@ asdf plugin add nodejs https://github.com/asdf-vm/asdf-nodejs.git
+### Updating node-build definitions
 
-## Problems with OpenPGP signatures in older versions
+Every new node version needs to have a definition file in the `node-build` repository. Because of that, `asdf-nodejs` tries to update the local `node-build` repository at every install command to download definitions for newly added versions.
 
-The plugin automatically imports the NodeJS release team's OpenPGP keys. If you are trying to install a previous release and facing any issue about verification, import the Node.js previous release team's OpenPGP keys to main keyring:
+We provide a command for manually updating `node-build` at your own time:
 
 ```bash
-bash -c '${ASDF_DATA_DIR:=$HOME/.asdf}/plugins/nodejs/bin/import-previous-release-team-keyring'
+asdf nodejs update-nodebuild
 ```
 
-## Using a dedicated OpenPGP keyring
+### Running the wrapped node-build command
 
-The `bash` script mentioned in [the installation instructions](#install) (`import-release-team-keyring`) imports the OpenPGP public keys in your main OpenPGP keyring. However, you can also use a dedicated keyring in order to mitigate [this issue](https://github.com/nodejs/node/issues/9859).
-
-To use a dedicated keyring, prepare the dedicated keyring and set it as the default keyring in the current shell:
+We provide a command for running the installed `node-build` command:
 
 ```bash
-export GNUPGHOME="${ASDF_DIR:-$HOME/.asdf}/keyrings/nodejs" && mkdir -p "$GNUPGHOME" && chmod 0700 "$GNUPGHOME"
-
-# Imports Node.js release team's OpenPGP keys to the keyring
-bash ~/.asdf/plugins/nodejs/bin/import-release-team-keyring
+asdf nodejs nodebuild --version
 ```
 
-Again, if you used `brew` to manage the `asdf` installation use the following bash commands:
+### node-build advanced variations
 
-```bash
-export GNUPGHOME="bash /usr/local/opt/asdf/keyrings/nodejs" && mkdir -p "$GNUPGHOME" && chmod 0700 "$GNUPGHOME"
+`node-build` has some additional variations aside from the versions listed in `asdf list-all nodejs` (chakracore/graalvm branches and some others). As of now, we weakly support these variations. In the sense that they are available for install and can be used in a `.tool-versions` file, but we don't list them as installation candidates nor give them full attention.
+
+Some of them will work out of the box, and some will need a bit of investigation to get them built. We are planning in providing better support for these variations in the future.
 
-# Imports Node.js release team's OpenPGP keys to the keyring
-bash /usr/local/opt/asdf/plugins/nodejs/bin/import-release-team-keyring
+To list all the available variations run:
+
+``` bash
+asdf nodejs nodebuild --definitions
 ```
 
-#### Related notes
+*Note that this command only lists the current `node-build` definitions. You might want to [update the local `node-build` repository](#updating-node-build-definitions) before listing them.*
```

Comment:
What if we updated `node-build` before listing the versions? We update kerl before listing Erlang versions in the Erlang plugin.

Reply:
We are not showing these versions on `asdf list-all` on this PR because the original `node-build` listing brings too many weird variations (some of them are even unsupported in some architectures), so I preferred to keep listing only the versions at https://nodejs.org/dist/.

I think we should move to `node-build` listing in a future PR